src/app/shared/components/delete-modal/delete-modal.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, EventEmitter, Output, ViewChild } from '@angular/core';
+import { NgbModal, NgbModalOptions, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { TestListItem } from '../../interfaces/test-list-item';
+
+@Component({
+  selector: 'app-delete-modal',
+  standalone: true,
+  imports: [],
+  templateUrl: './delete-modal.component.html',
+  styleUrl: './delete-modal.component.css',
+})
+export class DeleteModalComponent {
+  @ViewChild('modal') modal!: NgbModal;
+  protected activeModal?: NgbModalRef;
+  @Output() confirmDeleteEvent = new EventEmitter<boolean>();
+  reports: TestListItem[] = [];
+  deleteQuestion!: string;
+  deleteAllReports!: boolean;
```

Comment:
Both of these should allow being undefined. It is in general a good practice to allow it if there's a possibility. For the deleteAllReports being undefined would still be falsy, so it would still be false if put in an if-statement. For the deleteQuestion it might be better to assign either  assign an empty string to it or make the template wait for it to have a value by putting an @if statement around the html.
```suggestion
  deleteQuestion?: string;
  deleteAllReports?: boolean;
```

Reply:
Added the change. Somehow I didn't have to use an if-statement around the html. Is it a good idea to do it anyway?